packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormNumberFieldInput.tsx

```diff
@@ -0,0 +1,61 @@
+import { FormFieldInputBase } from '@/object-record/record-field/form-types/components/FormFieldInputBase';
+import { TextInput } from '@/ui/field/input/components/TextInput';
+import styled from '@emotion/styled';
+import { useState } from 'react';
+import {
+  canBeCastAsNumberOrNull,
+  castAsNumberOrNull,
+} from '~/utils/cast-as-number-or-null';
+
+const StyledInput = styled(TextInput)`
+  padding: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)}`};
+`;
+
+type FormNumberFieldInputProps = {
+  label?: string;
+  placeholder: string;
+  defaultValue: number | string | undefined;
+  onPersist: (value: number | null | string) => void;
+};
+
+export const FormNumberFieldInput = ({
+  label,
+  placeholder,
+  defaultValue,
+  onPersist,
+}: FormNumberFieldInputProps) => {
+  const [draftValue, setDraftValue] = useState(defaultValue ?? '');
+
+  const persistNumber = (newValue: string) => {
+    if (!canBeCastAsNumberOrNull(newValue)) {
+      return;
+    }
+
+    const castedValue = castAsNumberOrNull(newValue);
+
+    onPersist(castedValue);
+  };
+
+  const handleChange = (newText: string) => {
+    setDraftValue(newText);
+
+    persistNumber(newText.trim());
```

Comment:
Should we persist every time we change or could it be triggered by onBlur for example ? Or enter / esc ? This is maybe more product related.

Reply:
Thanks for mentioning this, @lucasbordeau. Currently, we persist in every change with some debouncing in the middle. We asked ourselves whether we had to implement the same blur and key listeners as for FieldInputs. However, we thought it was unnecessary as workflows don't work that way yet. We can revisit that decision later.